### PR TITLE
Implement TeleporterMixin.bridge$placeEntity for SpongeVanilla.

### PR DESCRIPTION
**Pull request by JBYoshi**
Got an AbstractMethodError for this while doing some testing. I just copied over the code from the Forge mixin and removed all the code for "isVanilla == false". Not sure whether this is supposed to be implemented on vanilla (the method was designed for Forge) or whether I just hit a bug.

Optional companion PR: https://github.com/SpongePowered/SpongeForge/pull/3017

### CODE REVIEW

File: src/main/java/org/spongepowered/common/mixin/core/world/TeleporterMixin.java [hunk 1]
```diff
@@ -82,6 +85,27 @@ public void placeInPortal(Entity entityIn, float rotationYaw) {
...
     }
 
+    @Override
+    public void bridge$placeEntity(net.minecraft.world.World world, Entity entity, float yaw) {
+        boolean didPort;
+        if (entity instanceof EntityPlayerMP) {
+            this.placeInPortal(entity, yaw);
+            didPort = true;
+        } else {
+            if (((WorldServerBridge) this.world).bridge$getDimensionId() == 1) {
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/mixin/core/world/TeleporterMixin.java]: Is it guaranteed the world is a `WorldServer` instance?

File: src/main/java/org/spongepowered/common/mixin/core/world/TeleporterMixin.java [hunk 2]
```diff
@@ -82,6 +85,27 @@ public void placeInPortal(Entity entityIn, float rotationYaw) {
...
+                didPort = true;
+            } else {
+                didPort = this.placeInExistingPortal(entity, yaw);
+            }
+        }
+        
+        if (PhaseTracker.getInstance().getCurrentContext() instanceof InvokingTeleporterContext) {
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/mixin/core/world/TeleporterMixin.java]: either replace with a `isTeleporting()` or make it a state based method that's overridden in the state, much like `isCapturing` or `isTicking`.

**Review by gabizou — CHANGES_REQUESTED** [review]
PhaseContext instanceof and casting, would be easier to make it on the state itself in my opinion.

File: src/main/java/org/spongepowered/common/mixin/core/world/TeleporterMixin.java (hunk 2 above)
Comment (Zidane) [reply src/main/java/org/spongepowered/common/mixin/core/world/TeleporterMixin.java]: That code comes from SpongeCommon but you don't have a bad point.

File: src/main/java/org/spongepowered/common/mixin/core/world/TeleporterMixin.java (hunk 1 above)
Comment (Zidane) [reply src/main/java/org/spongepowered/common/mixin/core/world/TeleporterMixin.java]: Yes

**Comment by JBYoshi** [comment]
@gabizou Updated.

File: src/main/java/org/spongepowered/common/event/tracking/IPhaseState.java [hunk 3]
```diff
@@ -906,4 +906,7 @@ default boolean isConvertingMaps() {
     default boolean allowsGettingQueuedRemovedTiles() {
         return false;
     }
+
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/event/tracking/IPhaseState.java]: Can you document what this method is intending to do? I know that not all of the methods in `IPhaseState` are well documented, but it's best when adding new ones that they are documented.

File: src/main/java/org/spongepowered/common/mixin/core/world/TeleporterMixin.java (hunk 1 above)
Comment (gabizou) [reply src/main/java/org/spongepowered/common/mixin/core/world/TeleporterMixin.java]: Make the id a constant (Even though we know the end dimension id is `1` for all of 1.12's lifetime.

**Review by gabizou — CHANGES_REQUESTED** [review]
Small documentation request and constant change. Otherwise, fine to merge.

**Pull request merged**